src/jwks.cc

```diff
@@ -493,16 +484,32 @@ void Jwks::createFromPemCore(const std::string& pkey_pem) {
       key_ptr->ec_key_.reset(EVP_PKEY_get1_EC_KEY(evp_pkey.get()));
       key_ptr->kty_ = "EC";
       break;
-    case EVP_PKEY_ED25519:
-      key_ptr->okp_key_ = std::move(evp_pkey);
+    case EVP_PKEY_ED25519: {
+      uint8_t raw_key[ED25519_PUBLIC_KEY_LEN];
+      size_t out_len = ED25519_PUBLIC_KEY_LEN;
+      if (EVP_PKEY_get_raw_public_key(evp_pkey.get(), raw_key, &out_len) != 1 ||
+          out_len != ED25519_PUBLIC_KEY_LEN) {
+        updateStatus(Status::JwksPemGetRawEd25519Error);
+        return;
+      }
+      key_ptr->okp_key_raw_ = (char*)raw_key;
```

Comment:
if you do this: it is null terminated string.  it will break if your raw_key has 0 in between,

you can do this:
key_ptr->okp_key_raw_ = std::string(static_cast<const char*>(raw_key), out_len);

Reply:
I kinda figured it was wrong and hoped you'd set me straight! :)

Reply:
I get this, looks like reinterpret_cast is required?

```
src/jwks.cc:496:23: error: static_cast from 'uint8_t *' (aka 'unsigned char *') to 'const char *' is not allowed
          std::string(static_cast<const char*>(raw_key), out_len);
```

Reply:
Yes, we need  reinterpret_cast